format.ts

```diff
@@ -1,36 +1,76 @@
-#!/usr/bin/env deno --allow-run
+#!/usr/bin/env deno --allow-run --allow-write
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import "./prettier/prettier.js";
+import "./prettier/parser-typescript.js";
+import "./prettier/parser-markdown.js";
+import { platform, readAll, exit, run, readFile, writeFile } from "deno";
 
-import { readAll, exit, run } from "deno";
+// TODO: provide decent type declarions for these
+const { prettier, prettierPlugins } = window as any;
 
-async function checkVersion() {
-  const prettierVersion = run({
-    args: ["bash", "-c", "prettier --version"],
-    stdout: "piped"
+// Runs commands in cross-platform way
+function xrun(opts) {
+  return run({
+    ...opts,
+    args: platform.os === "win" ? ["", "/c", ...opts.args] : opts.args
   });
-  const b = await readAll(prettierVersion.stdout);
-  const s = await prettierVersion.status();
-  if (s.code != 0) {
-    console.log("error calling prettier --version error");
-    exit(s.code);
-  }
-  const version = new TextDecoder().decode(b).trim();
-  const requiredVersion = "1.15";
-  if (!version.startsWith(requiredVersion)) {
-    console.log(`Required prettier version: ${requiredVersion}`);
-    console.log(`Installed prettier version: ${version}`);
-    exit(1);
+}
+
+const decoder = new TextDecoder();
+const encoder = new TextEncoder();
+
+function decode(data) {
+  return decoder.decode(data);
+}
+
+function encode(str) {
+  return encoder.encode(str);
+}
```

Comment:
I think it's more trouble than it's worth

Reply:
I don't see what exactly you mean. Do you think inlining it like `new TextEncoder().encode(...)` is better?

Reply:
inlining `encoder.encode(str)` isn't much more to write than `encode(str)` - and it prevents two extra functions